msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/states/AccountState.kt

```diff
@@ -220,21 +237,72 @@ class AccountState private constructor(
     }
 
     /**
-     * Retrieves the access token for the currently signed in account from the cache.
+     * Retrieves the access token for the currently signed in account from the cache. If multiple
+     * access tokens are present in the cache then the one that matches the sign in scopes is returned.
      * If the access token is expired, it will be attempted to be refreshed using the refresh token that's stored in the cache;
      * Kotlin coroutines variant.
      *
      * @return [com.microsoft.identity.nativeauth.statemachine.results.GetAccessTokenResult] The result of the getAccessToken action
      */
+    @Deprecated("Use the getAccessToken(forceRefresh: Boolean = false, scopes: List<String>) method")
     suspend fun getAccessToken(forceRefresh: Boolean = false): GetAccessTokenResult {
+        return getAccessTokenInternal(forceRefresh, null);
+    }
+
+    /**
+     * Retrieves the access token for the currently signed in account from the cache.
+     * If the access token is expired, it will be attempted to be refreshed using the refresh token that's stored in the cache;
+     * If any of the cached access token matches the scope specified in scopes parameter then that
+     * is returned otherwise a new access token with requested scopes is fetched.
+     * Kotlin coroutines variant.
+     *
+     * @return [com.microsoft.identity.nativeauth.statemachine.results.GetAccessTokenResult] The result of the getAccessToken action
+     */
+    suspend fun getAccessToken(forceRefresh: Boolean = false, scopes: List<String>?): GetAccessTokenResult {
+        return getAccessTokenInternal(forceRefresh, scopes)
+    }
+
+    /**
+     * Retrieves the access token for the currently signed in account from the cache.
+     * If the access token is expired, it will be attempted to be refreshed using the refresh token that's stored in the cache;
+     * If any of the cached access token matches the scope specified in scopes parameter then that
+     * is returned otherwise a new access token with requested scopes is fetched.
+     * callback variant.
+     *
+     * @return [com.microsoft.identity.client.IAuthenticationResult] If successful.
+     * @throws [MsalClientException] If the the account doesn't exist in the cache.
+     * @throws [ServiceException] If the refresh token doesn't exist in the cache/is expired, or the refreshing fails.
+     */
+    fun getAccessToken(forceRefresh: Boolean = false, scopes: List<String>, callback: GetAccessTokenCallback) {
```

Comment:
Scopes param is not nullable, unlike it's coroutine sibling method. Please align the two.

Reply:
Done